### PR TITLE
Bug fix.

### PR DESCRIPTION
**Pull request by vel21ripn**
Restore part of the fields flow->packet after cleaning flow structure.

### CODE REVIEW

File: src/lib/ndpi_main.c [hunk 1]
```diff
@@ -4249,11 +4249,35 @@ static int ndpi_init_packet_header(struct ndpi_detection_module_struct *ndpi_str
...
+	u_int64_t current_time_ms;
+	u_int16_t l3_packet_len;
+	u_int16_t l4_packet_len;
+	u_int16_t payload_packet_len;
+	u_int16_t actual_payload_len;
+	u_int8_t l4_protocol;
+
+#define packet_save(a) a = flow->packet.a
+#define packet_restore(a) flow->packet.a = a
+
```
Comment (IvanNardi) [inline src/lib/ndpi_main.c]: Is it feasible to use these macros for the fields we already restore, too? (guessed_protocol_id, num_processed_pkts....)
This way the code should be more uniform

**Comment by vel21ripn** [comment]
It is quick and dirty. This is one of the possible options for correcting the error.
If you can offer a more elegant option, I close my PR.

**Comment by IvanNardi** [comment]
> It is quick and dirty. This is one of the possible options for correcting the error.
> If you can offer a more elegant option, I close my PR.

Sorry, I have not been clear.
I like your solution! I just wonder if you can extend it to all the fields we restore (instead of using these macros only for the "new" fields and the old code for the "old"fields).
Anyway, your PR is ok, IMO!

**Comment by vel21ripn** [comment]
New, more uniform version.

**Pull request merged**